src/auth/session/session-utils.ts

```diff
@@ -9,23 +9,49 @@ export function sessionFromEntries(
   const obj = Object.fromEntries(
     entries
       .filter(([_key, value]) => value !== null)
+      // Sanitize keys
       .map(([key, value]) => {
         switch (key.toLowerCase()) {
           case 'isonline':
             return ['isOnline', value];
           case 'accesstoken':
             return ['accessToken', value];
+          case 'onlineaccessinfo':
+            return ['onlineAccessInfo', value];
           default:
             return [key.toLowerCase(), value];
         }
+      })
+      // Sanitize values
+      .map(([key, value]) => {
+        switch (key) {
+          case 'isOnline':
+            if (typeof value === 'string') {
+              return [key, value.toString().toLowerCase() === 'true'];
+            } else if (typeof value === 'number') {
+              return [key, Boolean(value)];
+            }
+            return [key, value];
+          case 'scope':
+            return [key, value.toString()];
+          case 'expires':
+            return [key, new Date(Number(value) * 1000)];
```

Comment:
This is potentially null, do we need to handle that here?

Reply:
Good catch! Added test and fix